# Prosjektoppgave - statistisk kvalitetskontroll

Mål med oppgaven:
* Deskriptiv statistikk
* Lage Shewhart- og s-diagram
* Tolkning av kapabilitetsindeks

En bedrift er ansvarlig for sløying og pakking av fisk og ønsker å foreta kontrollmålinger for å sikre at vekten av hver fiskefilet de selger til kundene sine er korrekt. Anta at bedriften hver halvtime gjennomfører en stikkprøve der de veier $n = 5$ tilfeldige valgte fiskefileter. Bedriftene gjør $k = 10$ stikkprøver. 

Resultatene er oppgitt i hver tabellen der hver rad i tabellen svarer til en stikkprøve. Vekten er oppgitt i gram.

In [2]:
n = 5
k = 10

tot = n * k

mu0 = 100
sigma0 = 1

import numpy as np
import pandas as pd

import matplotlib.pyplot as plt
%matplotlib inline

#pip install tabulate
np.random.seed(seed=1)

df = pd.DataFrame(mu0 + sigma0 * np.random.randn(k, n))

from IPython.display import HTML, display
display(df)


0           1           2           3           4
0  101.624345   99.388244   99.471828   98.927031  100.865408
1   97.698461  101.744812   99.238793  100.319039   99.750630
2  101.462108   97.939859   99.677583   99.615946  101.133769
3   98.900109   99.827572   99.122142  100.042214  100.582815
4   98.899381  101.144724  100.901591  100.502494  100.900856
5   99.316272   99.877110   99.064231   99.732112  100.530355
6   99.308339   99.603246   99.312827   99.154794   99.328754
7   99.987335   98.882690  100.234416  101.659802  100.742044
8   99.808164   99.112371   99.252842  101.692455  100.050808
9   99.363004  100.190915  102.100255  100.120159  100.617203

a)

Visualiser dataene. 

Diskuter kort hva du ser.

In [ ]:
# Svar på a)

b)

Finn gjennomsnitt og standardavvik i hvert av de $k$ forsøkene. 

Finn tallverdier for nedre og øvre kontrollgrense når $z_{\alpha / 2} =3$.

Finn også nedre og øvre varselgrense.

Bedriften ønsker å selge fileter med vekt $\mu_0 = 100$ gram. Finn minste antall kontroller bedriften må forvente å gjennomføre for å oppdage at vekten har endret set til $\mu_1 = 99,5$ gram. 

In [ ]:
# Svar på b) 

Bedriften har bestemt seg for å gjennomføre $m = 10$ nye stikkeprøver neste dag gjennomført som før. Resultatet av disse gitt i tabellen under. 

In [5]:
k_ny = 10

np.random.seed(seed=2)
df_ny = pd.DataFrame(mu0 -.5 + sigma0 * 2 * np.random.randn(k_ny, n))
display(df_ny)



0           1           2           3           4
0   98.666484   99.387466   95.227608  102.780542   95.913129
1   97.816505  100.505763   97.009424   97.384096   97.681985
2  100.602908  104.084416   99.583079   97.264149  100.578117
3   98.307681   99.461739  101.850002   98.004258   99.518051
4   97.743784   99.187132  100.013141   97.522442   98.822356
5   99.027632   98.224690   97.124775   96.657566   99.193010
6   98.961886  103.962734   94.630465   99.725453  100.240889
7  102.219268  100.503714   97.811573   99.500020  100.584705
8   98.872984  101.042023   95.763819  102.962369  102.435356
9   98.828645  100.722682   99.595941   97.841729   99.675420

c)

Bedriften ønsker å kontrollere om det har inntruffet et kvalitetsavvik. Formuler situasjonen som en hypotese.

Bruk kontrollgrensene du fant over og lag et Shewhart $\bar{x}$-diagram for de neste $m = 10$ observasjonene. 

Diskuter hva du ser. 

In [ ]:
# Svar på c)

d)

Lag et $s$-diagram for spredning. 

Spesifiser hypotesen.

En butikk kjøper fisk av bedriften. Butikken har stilt følgende krav om toleransegrense til produsenten: $T_L = 97$ og $T_u = 103$. 

Beregn kapabilitet og kapalitetsindeks.

Tolk kort kapabilitetsindeksen.

In [ ]:
# Svar på d)